HARK/distribution.py

```diff
@@ -1196,9 +1227,27 @@ def combine_indep_dstns(*distributions, seed=0):
     assert np.isclose(np.sum(P_out), 1), "Probabilities do not sum to 1!"
     return DiscreteDistribution(P_out, X_out, seed=seed)
 
-def calc_expectation(dstn,func=lambda x : x,*args):
+
+# # 20210619: CDC: This method for taking expectations is designed for efficiency but is
+# # not at all intuitive for the user because it works only with functions that take as
+
+# handcrafted specialized functions that know that their positional first argument must be
+# a matrix whose structure matches the structure of the discrete income distribution provided
+# as the first argument. For example, if the 2D distribution is of transitory and permanent
+# shocks, the function must take as an input a matrix which is assumed to have a structure
+# that exactly mirrors the internal structure of the arrangement of broadcasted transitory
+# and permanent shocks inside the dstn object, and knows(for example) that the value of the permanent shock can be how to extract
+
+# hard to understand because it requires the user to
+# # provide an array of values of its inputs but then assumes it already knows what those
+# # inputs are. This makes little sense. If it is going to assume it knows what the inputs
```

Comment:
I don't think this comment should be included in the code as is.

It has strange formatting/grammatical errors.

Also, it's not clear what is gained by having this critical opinion included in the source code. It is distracting and, as an opinion, contestable.

It would be more appropriate to raise this as an issue or discussion on GitHub. It is quite possible that there is a better design for the function. The source code comments are not the right place to debate that design.

Reply:
I've put comments of this kind pervasively throughout the revisions I've been making, mainly to explain them to the reviewers (you, Mridul, Matt) upon review, and with the expectation that the right workflow will be, if the comment's substance is something we agree on, it should be formulated as an issue or a "TODO" item.  

You are welcome to begin that process right now, with this comment, if you like.  (The odd formatting is just because it seems I commented the inner block earlier and later commented a larger block containing the inner comment at some point, leading to two #)

Reply:
Ok. So do I understand correctly that you would like Mridul or I to take this PR and make whatever modifications to it are necessary to make it pass review, including:
 - fixing the docstring location 
 - removing these comments in distribution.py but turning them into issues?

Reply:
Yes, that's exactly what I had in mind.  For anything that does not change the functioning of any code, you should propose whatever changes (to docstrings, comments, code formatting or structure, etc) you think suitable here, and I'll either approve them (likely) or follow up with a response.

In this case, just fixing the docstring will not require a response.

Reply:
Ok. I see that this a branch on the main HARK repository so I can make these changes myself with no permission conflicts.
I'll get to this over the next few days.